Mathlib/LinearAlgebra/TensorProduct/RightExactness.lean

```diff
@@ -0,0 +1,358 @@
+      Submodule.Quotient.mk (p := LinearMap.range (rTensor Q f)) y := by
+  simp only [← LinearMap.comp_apply, ← Submodule.mkQ_apply]
+  rw [exact_iff] at hfg
+  apply LinearMap.congr_fun
+  apply TensorProduct.ext'
+  intro n q
+  simp only [coe_comp, Function.comp_apply, rTensor_tmul, Submodule.mkQ_apply]
+  simp only [rTensor.inverse_of_rightInverse]
+  simp only [TensorProduct.lift.tmul, coe_mk, AddHom.coe_mk, mk₂_apply]
+  simp only [coe_comp, Function.comp_apply, mk_apply, Submodule.mkQ_apply]
+  rw [Submodule.Quotient.eq, ← TensorProduct.sub_tmul]
+  apply le_comap_range_rTensor f
+  rw [← hfg, mem_ker, map_sub, sub_eq_zero, hgh]
+
+/-- The inverse map in `rTensor.equiv` -/
+noncomputable
+def rTensor.inverse :
+    P ⊗[R] Q →ₗ[R] N ⊗[R] Q ⧸ LinearMap.range (rTensor Q f) :=
+  rTensor.inverse_of_rightInverse Q hfg (Function.rightInverse_surjInv hg)
+
+lemma rTensor.inverse_apply (y : N ⊗[R] Q) :
+    (rTensor.inverse Q hfg hg) ((rTensor Q g) y) =
+      Submodule.Quotient.mk (p := LinearMap.range (rTensor Q f)) y := by
+  rw [rTensor.inverse, rTensor.inverse_of_rightInverse_apply]
+
+lemma rTensor.inverse_comp_rTensor :
+    (rTensor.inverse Q hfg hg).comp (rTensor Q g) =
+      Submodule.mkQ (p := LinearMap.range (rTensor Q f)) := by
+  rw [LinearMap.ext_iff]
+  intro y
+  simp only [coe_comp, Function.comp_apply, Submodule.mkQ_apply, rTensor.inverse_apply]
+
+/-- For a surjective `f : N →ₗ[R] P`,
+  the natural equivalence between `N ⊗[R] Q ⧸ (range (rTensor Q f))` and `P ⊗[R] Q`
+  (computably, given a right inverse) -/
+def rTensor.equiv_of_rightInverse
+    {h : P → N} (hgh : Function.RightInverse h g) :
+    ((N ⊗[R] Q) ⧸ (range (rTensor Q f))) ≃ₗ[R] (P ⊗[R] Q) := {
+  rTensor.toFun Q hfg with
+  invFun    := rTensor.inverse_of_rightInverse Q hfg hgh
+  left_inv  := fun y ↦ by
+    simp only [rTensor.toFun, AddHom.toFun_eq_coe, coe_toAddHom]
+    obtain ⟨y, rfl⟩ := Submodule.mkQ_surjective _ y
+    simp only [Submodule.mkQ_apply, Submodule.liftQ_apply, rTensor.inverse_of_rightInverse_apply]
+  right_inv := fun z ↦ by
+    simp only [AddHom.toFun_eq_coe, coe_toAddHom]
+    obtain ⟨y, rfl⟩ := rTensor.surjective Q hgh.surjective z
+    rw [rTensor.inverse_of_rightInverse_apply]
+    simp only [rTensor.toFun, Submodule.liftQ_apply] }
+
+/-- For a surjective `f : N →ₗ[R] P`,
+  the natural equivalence between `N ⊗[R] Q ⧸ (range (rTensor Q f))` and `P ⊗[R] Q` -/
+noncomputable def rTensor.equiv :
+    ((N ⊗[R] Q) ⧸ (LinearMap.range (rTensor Q f))) ≃ₗ[R] (P ⊗[R] Q) :=
+  rTensor.equiv_of_rightInverse Q hfg (Function.rightInverse_surjInv hg)
+
+/-- Tensoring an exact pair on the right gives an exact pair -/
+theorem rTensor_exact : Exact (rTensor Q f) (rTensor Q g) := by
+  rw [exact_iff]
+  rw [← Submodule.ker_mkQ (p := range (rTensor Q f))]
+  rw [← rTensor.inverse_comp_rTensor Q hfg hg]
+  apply symm
+  apply ker_comp_of_ker_eq_bot
+  rw [ker_eq_bot]
+  exact (rTensor.equiv Q hfg hg).symm.injective
```

Comment:
```suggestion
theorem rTensor_exact : Exact (rTensor Q f) (rTensor Q g) := by
  rw [exact_iff, ← Submodule.ker_mkQ (p := range (rTensor Q f)),
    ← rTensor.inverse_comp_rTensor Q hfg hg]
  apply symm
  apply ker_comp_of_ker_eq_bot
  rw [ker_eq_bot]
  exact (rTensor.equiv Q hfg hg).symm.injective
```

Reply:
why not, but I'm not convinced. Done, anyway.